src/main/java/org/skife/jdbi/v2/StatementContext.java

```diff
@@ -55,6 +57,14 @@
      */
     Map<String, Object> getAttributes();
 
+    /**
+     * Obtain a column mapper for the given type in this context.
+     *
+     * @param type the target type to map to
+     * @return a ResultColumnMapper for the given type
+     * @throws org.skife.jdbi.v2.exceptions.DBIException if no mappers are registered for the given type.
+     */
+    ResultColumnMapper columnMapperFor(Class type);
```

Comment:
I am pretty sure that this is not strong enough. This should be `Type` so that it is possible to pass in e.g. parameterized types. Interfaces that take `Class<?>` almost always needs retrofitting for later Java versions.